src/passes/J2CLOpts.cpp

```diff
@@ -30,9 +30,48 @@ namespace wasm {
 
 namespace {
 
+using AssignmentCountMap = std::unordered_map<Name, Index>;
+using TrivialFunctionMap = std::unordered_map<Name, Expression*>;
+
 bool isOnceFunction(Function* f) { return f->name.hasSubstring("_<once>_"); }
 
-using AssignmentCountMap = std::unordered_map<Name, Index>;
+// Returns the function body if it is a trivial function, null otherwise.
+Expression* getTrivialFunctionBody(Function* func) {
+  auto* body = func->body;
+
+  // Only consider trivial the following instructions which can be safely
+  // inlined and note that their size is at most 2.
+  if (body->is<Nop>() || body->is<GlobalGet>() || body->is<Const>() ||
+      (body->is<Call>() && body->dynCast<Call>()->operands.size() == 0) ||
+      (body->is<GlobalSet>() &&
+       body->dynCast<GlobalSet>()->value->is<Const>())) {
```

Comment:
Were we covering `global_foo = global_bar`? I guess that was more than size 2; though maybe it is worthwhile?

Reply:
If `global_bar` becomes immutable it is handle by the propagate. But `(global.set $a (global.get $b)`  is still size 2 isn't it?.  Let me log and see if there are any ocurrences of this and if we should in general worry about it.

Reply:
Tested both scenarios `global.set ... (global.get....)` and only inlining `$$class-initialized` and does not seem to have any impact in either our size report nor in the bigger apps.